components/ui/form/Schedule.tsx

```diff
@@ -0,0 +1,189 @@
+import { PlusIcon, TrashIcon } from "@heroicons/react/outline";
+import dayjs, { Dayjs, ConfigType } from "dayjs";
+import React, { useCallback, useState } from "react";
+import { Controller, useFieldArray, useFormContext } from "react-hook-form";
+
+import { weekdayNames } from "@lib/core/i18n/weekday";
+import { useLocale } from "@lib/hooks/useLocale";
+import { TimeRange, Schedule as ScheduleType } from "@lib/types/schedule";
+
+import Button from "@components/ui/Button";
+import Select from "@components/ui/form/Select";
+
+/** Begin Time Increments For Select */
+const increment = 15;
+/**
+ * Creates an array of times on a 15 minute interval from
+ * 00:00:00 (Start of day) to
+ * 23:45:00 (End of day with enough time for 15 min booking)
+ */
+const TIMES = (() => {
+  const end = dayjs().endOf("day");
+  let t: Dayjs = dayjs().startOf("day");
+
+  const times = [];
+  while (t.isBefore(end)) {
+    times.push(t);
+    t = t.add(increment, "minutes");
+  }
+  return times;
+})();
+/** End Time Increments For Select */
+
+const defaultDayRange: TimeRange = {
+  start: dayjs("1970-01-01T09:00:00").valueOf(),
+  end: dayjs("1970-01-01T17:00:00").valueOf(),
+};
+
+export const DEFAULT_SCHEDULE: ScheduleType = [
+  [],
+  [defaultDayRange],
+  [defaultDayRange],
+  [defaultDayRange],
+  [defaultDayRange],
+  [defaultDayRange],
+  [],
+];
+
+type Option = {
+  readonly label: string;
+  readonly value: number;
+};
+
+type TimeRangeFieldProps = {
+  name: string;
+};
+
+const TimeRangeField = ({ name }: TimeRangeFieldProps) => {
```

Comment:
Are you planning on changing these to be more RHF-esque? I'm ok either way, don't have to do everything at once

Reply:
Let's leave it for now, reason being is that currently name ties into RHF - what I noticed is that if I wanted to use register throughout the component I ended up adding a load of code for no reason - because RHF is already available throughout. In the end the leanest approach here is to useFormContext and use it where it is actually used without passthrough.